### DISCUSSION TITLE
Connect to postgres with ssl mode

### DISCUSSION THREAD

**Discussion by tagurus** [sequelize/sequelize#15636] [Help & Questions] [answered]
Greetings!

I'm using sequelize v. 5.22.4. When I'm trying to connect to the postgres database and specifying certificates in dialectOptions, an error appears: "self signed certificate in certificate chain". I've tried setting the rejectUnauthorized: false parameter, but it does not give any result, because any record or change of data in the database still gives the same error. Certificates is valid, signed by CA.

Connection code:

sequelize = new Sequelize(url, {
      dialect: 'postgres',
      dialectOptions: {
        ssl: {
          rejectUnauthorized: false,
          ca: fs.readFileSync(caPath).toString(),
          key: fs.readFileSync(keyPath).toString(),
          cert: fs.readFileSync(certPath).toString(),
        },
      }
    })

We also tried to set different sslmode in connection url (required, verify-ca, etc.). 

Other DBMS (in our case navicat) work fine with our certificates, connection goes through, data are written.

**Comment by ephys** [comment, marked as answer]
Unfortunately I do not know. Everything in `dialectOptions` is sent as-is to the `pg` library, specifically to the `Client` constructor. Maybe something here can help you: https://node-postgres.com/apis/client

**Reply by jayce7** [reply]
I think there is a bug (but I'm a newbie so maybe I'm wrong).  When I test my connection with pg directly, I can connect to my Google Cloud PG DB over SSL.  But when I port the same code over to my config.js, I get weird behavior.  It doesn't seem to like getting values from environment variables when using the Production environment.  I can use them for my Test environment, but when I complete the Production section of Config.JS with the process.env.DB_NAME etc., it tries to authenticate with my computer's username (i.e. Jason).  When I put a string in for username as 'testaccount' but left the database pulling its values from .env, I got the error... ERROR: database "testaccount" does not exist.  But when I put the string directly in for database in the config.js, it worked.  I think some mappings or variable switching could be going on?
